### DISCUSSION TITLE
服务端是 docker, 无法刮削，请帮看看是什么问题？

### DISCUSSION THREAD

**Discussion by aleylinx** [metatube-community/jellyfin-plugin-metatube#233] [Q&A]
jellyfin和metatube的服务端都是安装在qnap上的，用　ｄｏｃｋｅｒ安装的。
jellyfin的IP是10.0.3.2，而这个metatube的ip是 10.0.3.3，无法刮削，不知道是什么原因？
NAS的IP是192.168.1.103
奇怪的是，我的wifi ip是 192.168.1.102 怎么也出现在这里面了？

打开日志，发现如下信息：

`[GIN] 2023/11/15 - 09:40:39 | 200 | 30.008601892s |        10.0.3.2 | GET      "/v1/movies/search?q=DLDSS-230&provider=&fallback=True"
[GIN] 2023/11/15 - 09:40:39 | 404 |      48.029µs |        10.0.3.2 | GET      "/v1/movies?lazy=True"
[GIN] 2023/11/15 - 09:40:39 | 200 |      59.295µs | 192.168.1.102 | GET      "/"
[GIN] 2023/11/15 - 09:40:40 | 404 |      42.652µs | 192.168.1.102 | GET      "/favicon.ico"
[GIN] 2023/11/15 - 09:41:26 | 500 |  6.211661475s |        10.0.3.2 | GET      "/v1/actors/GFriends/夏川あゆみ?lazy=True"
[GIN] 2023/11/15 - 09:44:32 | 404 |      84.664µs |        10.0.3.2 | GET      "/v1/actors/GFriends/夏川あゆみ?lazy=True"
[GORM] 2023/11/15 09:46:36 github.com/metatube-community/metatube-sdk-go/engine/movie.go:24
[0.311ms] [rows:0] SELECT * FROM `movie_metadata` WHERE number = "DLDSS-239" COLLATE NOCASE OR id = "DLDSS-239" COLLATE NOCASE
[GIN] 2023/11/15 - 09:46:36 | 200 | 41.997051403s |        10.0.3.2 | GET      "/v1/movies/search?q=DLDSS-239&provider=&fallback=True"
[GIN] 2023/11/15 - 09:46:36 | 404 |      38.501µs |        10.0.3.2 | GET      "/v1/movies?lazy=True"
`

**Comment by aleylinx** [comment]
打开jellyfin的日志，看到如下信息：

> INF] [100] Jellyfin.Plugin.MetaTube.Providers.MovieProvider: Search for movie: DLDSS-239
> [11:13:54] [INF] [85] Jellyfin.Plugin.MetaTube.Providers.MovieProvider: Get movie info: :
> [11:13:54] [ERR] [94] MediaBrowser.Providers.Movies.MovieMetadataService: Error in MetaTube
> System.Exception: API request error: 404 (Not Found)
>    at Jellyfin.Plugin.MetaTube.ApiClient.GetDataAsync[T](String url, Boolean requireAuth, CancellationToken cancellationToken)
>    at Jellyfin.Plugin.MetaTube.ApiClient.GetMovieInfoAsync(String provider, String id, Boolean lazy, CancellationToken cancellationToken)
>    at Jellyfin.Plugin.MetaTube.ApiClient.GetMovieInfoAsync(String provider, String id, CancellationToken cancellationToken)
>    at Jellyfin.Plugin.MetaTube.Providers.MovieProvider.GetMetadata(MovieInfo info, CancellationToken cancellationToken)
>    at MediaBrowser.Providers.Manager.MetadataService`2.ExecuteRemoteProviders(MetadataResult`1 temp, String logName, TIdType id, IEnumerable`1 providers, CancellationToken cancellationToken)

**Comment by aleylinx** [comment]
据说  插件里的 provider filter 选项被打勾了  就无法刮削……

**Reply by Topbcy** [reply]
一旦打勾了這個 provider filter 選項，若沒有填上 Provider 的話，就不會刮削到任何東西

**Reply by xjasonlyu** [reply]
是的，filter就是你需要什么才出来什么，你什么都没填自然就什么都没有的。

**Reply by hejunchen** [reply]
我的理解是，如果你打勾了，那么代码就会用filter里的provider去筛选。如果你不填任何值，那么就等于不会找到任何provider去搜刮。我之前也是搞了这个，一直找不到数据。后来去掉勾就好了。

**Reply by xjasonlyu** [reply]
你说的有道理，主要是有时候大家不明白这个参数是啥意思就勾上了。我更新一下，当filter为空的时候就当作没设置好了。

**Reply by hejunchen** [reply]
是的，这是个 user flow 的问题。应当把空白的filter当作没有设定filter来处理。而且在log日志里应该标识出来选择的filter为空，至少这属于一个特殊值，如果是刻意为之，那么至少应当给出一定的提示。

**Reply by neoczr** [reply]
那其实要不要勾filter呢？
